### PR TITLE
[backport] Import DLPack header file & Fix multiple issues

### PR DESCRIPTION
**Pull request by emcastillo**
Backport of #4517

### CODE REVIEW

**Comment by emcastillo** [comment]
[automatic post] Jenkins, test this please.

File: cupy/core/include/cupy/README.md [hunk 1]
```diff
@@ -10,3 +10,13 @@ The `cub` folder is a git submodule for the CUB project.
 Including the CUB headers as a submodule enables not only building the `cupy.cuda.cub` module,
 but also easier maintenance.
 For further information on CUB, see the [CUB Project Website](http://nvlabs.github.com/cub).
+
+## Jitify
+The `Jitify` folder is a git submodule for the Jitify project.
+Including the Jitify header as a submodule for building the `cupy.cuda.jitify` module.
+For further information on Jitify, see the [Jitify repo](https://github.com/NVIDIA/jitify).
+
```
Comment (leofang) [inline cupy/core/include/cupy/README.md:18]: This should be removed as Jitify is not present in v8.
```suggestion
```

**Review by leofang — COMMENTED** [review]
Thanks for backporting it for me, @emcastillo! LGTM except for a minor comment.

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Pull request merged**